examples/threadExample/src/main/java/io/realm/examples/threads/AsyncQueryFragment.java

```diff
@@ -0,0 +1,190 @@
+        realm = Realm.getDefaultInstance();
+        mAllSortedDots = realm.where(Dot.class)
+                .between("x", 25, 75)
+                .between("y", 0, 50)
+                .findAllSortedAsync(
+                        "x", RealmResults.SORT_ORDER_ASCENDING,
+                        "y", RealmResults.SORT_ORDER_DESCENDING);
+        mDotAdapter.updateList(mAllSortedDots);
+        mAllSortedDots.addChangeListener(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        // Remember to close the Realm instance when done with it.
+        cancelTransaction();
+        mAllSortedDots.removeChangeListener(this);
+        mAllSortedDots = null;
+        realm.close();
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            case R.id.translate_button: {
+                cancelTransaction();
+                // translate all points coordinates using an async transaction
+                mAsyncTransaction = realm.executeTransaction(new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        // query for all points
+                        RealmResults<Dot> dots = realm.where(Dot.class).findAll();
+
+                        // Iterating backwards to avoid https://github.com/realm/realm-java/issues/640
+                        for (int i = dots.size() - 1; i >= 0; i--) {
+                            Dot dot = dots.get(i);
+                            if (dot.isValid()) {
+                                int x = dot.getX();
+                                int y = dot.getY();
+                                dot.setX(y);
+                                dot.setY(x);
+                            }
+                        }
+                    }
+                }, new Realm.Transaction.Callback() {
+                    @Override
+                    public void onSuccess() {
+                        if (isAdded()) {
+                            Toast.makeText(getActivity(), "Translation completed", Toast.LENGTH_SHORT).show();
+                        }
+                    }
+
+                    @Override
+                    public void onError(Exception e) {
+                        if (isAdded()) {
+                            Toast.makeText(getActivity(), "Error while translating dots", Toast.LENGTH_SHORT).show();
+                            e.printStackTrace();
+                        }
+                    }
+                });
+                break;
+            }
+        }
+    }
+
+    private void cancelTransaction() {
+        if (mAsyncTransaction != null && !mAsyncTransaction.isCancelled()) {
+            mAsyncTransaction.cancel();
+            mAsyncTransaction = null;
+        }
+    }
+
+    @Override
+    public void onChange() {
+        mDotAdapter.notifyDataSetChanged();
+    }
+
+
+    private class DotAdapter extends BaseAdapter {
```

Comment:
I guess you chose a normal BaseAdapter instead of RealmBaseAdapter to showcase how it is done, but perhaps add a note about RealmBaseAdapter and that if you use that you don't have to call notifyChange yourself?